project/cmake/scripts/darwin/macros.cmake

```diff
@@ -48,34 +48,51 @@ function(core_link_library lib wraplib)
 endfunction()
 
 function(find_soname lib)
-  if(ARGV1)
-    set(liblow ${ARGV1})
-  else()
-    string(TOLOWER ${lib} liblow)
-  endif()
+  string(TOLOWER ${lib} liblow)
   if(${lib}_LDFLAGS)
     set(link_lib "${${lib}_LDFLAGS}")
   else()
-    if(IS_ABSOLUTE "${${lib}_LIBRARIES}")
-      set(link_lib "${${lib}_LIBRARIES}")
-    else()
-      set(link_lib -l${${lib}_LIBRARIES})
-    endif()
+    set(link_lib "${${lib}_LIBRARIES}")
   endif()
-  execute_process(COMMAND ${CMAKE_C_COMPILER} -nostdlib -o /dev/null -Wl,-M ${link_lib} 
-                  COMMAND grep LOAD.*${liblow}
+
+  execute_process(COMMAND ${CMAKE_C_COMPILER} -print-search-dirs
+                  COMMAND fgrep libraries:
+                  COMMAND sed "s/[^=]*=\\(.*\\)/\\1/"
+                  COMMAND sed "s/:/ /g"
                   ERROR_QUIET
-                  OUTPUT_VARIABLE ${lib}_FILENAME)
-  string(REPLACE "LOAD " "" ${lib}_FILENAME "${${lib}_FILENAME}")
-  string(STRIP "${${lib}_FILENAME}" ${lib}_FILENAME)
-  if(${lib}_FILENAME)
-    execute_process(COMMAND objdump -p ${${lib}_FILENAME}
-                    COMMAND grep SONAME.*${liblow}
-                    ERROR_QUIET
-                    OUTPUT_VARIABLE ${lib}_SONAME)
-    string(REPLACE "SONAME " "" ${lib}_SONAME ${${lib}_SONAME})
-    string(STRIP ${${lib}_SONAME} ${lib}_SONAME)
-    message(STATUS "${lib} soname: ${${lib}_SONAME}")
-    set(${lib}_SONAME ${${lib}_SONAME} PARENT_SCOPE)
+                  OUTPUT_VARIABLE cc_lib_path
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+  execute_process(COMMAND echo ${link_lib}
+                  COMMAND sed "s/-L[ ]*//g"
+                  COMMAND sed "s/-l[^ ]*//g"
+                  ERROR_QUIET
+                  OUTPUT_VARIABLE env_lib_path
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if(CMAKE_CROSSCOMPILING)
+    set(host_lib_path /usr/lib /usr/local/lib)
   endif()
+
+  foreach(path ${cc_lib_path} ${env_lib_path} ${host_lib_path})
```

Comment:
I was not sure about that one either. I kept it as it was done that way in the configure script. If I'm not mistaken, CMAKE_CROSSCOMPILING is anyways not set in the Toolchain file so we'd never add the host paths. I'll remove that lines.